hypothesis-python/tests/django/toystore/forms.py

```diff
@@ -0,0 +1,208 @@
+
+from django import forms
+from django.forms import widgets
+
+from tests.django.toystore.models import (
+    CouldBeCharming,
+    Customer,
+    ManyNumerics,
+    ManyTimes,
+    OddFields,
+)
+
+
+class ReprModelForm(forms.ModelForm):
+    def __repr__(self):
+        """I recommend putting this in your form to show the failed cases."""
+        return "%r\n%r" % (self.data, self.errors)
+
+
+class ReprForm(forms.Form):
+    def __repr__(self):
+        return "%r\n%r" % (self.data, self.errors)
+
+
+class CouldBeCharmingForm(ReprModelForm):
+    class Meta:
+        model = CouldBeCharming
+        fields = "__all__"
+
+
+class CustomerForm(ReprModelForm):
+    class Meta:
+        model = Customer
+        fields = "__all__"
+
+
+class ManyNumericsForm(ReprModelForm):
+    class Meta:
+        model = ManyNumerics
+        fields = "__all__"
+
+
+class ManyTimesForm(ReprModelForm):
+    class Meta:
+        model = ManyTimes
+        fields = "__all__"
+
+
+class OddFieldsForm(ReprModelForm):
+    class Meta:
+        model = OddFields
+        fields = "__all__"
+
+
+class DynamicForm(ReprForm):
+    def __init__(self, field_count=5, **kwargs):
+        super(DynamicForm, self).__init__(**kwargs)
+        for i in range(field_count):
+            field_name = "field-%d" % (i,)
+            self.fields[field_name] = forms.CharField(required=False)
+
+
+class BasicFieldForm(ReprForm):
+    _boolean_required = forms.BooleanField()
+    _boolean = forms.BooleanField(required=False)
+    # This took me too long to figure out... The BooleanField will actually
+    # raise a ValidationError when it recieves a value of False. Why they
+    # didn't call it a TrueOnlyField escapes me, but *if* you actually want
+    # to accept both True and False in your BooleanField, make sure you set
+    # `required=False`. This behavior has been hotly contested in the bug
+    # tracker (e.g. https://code.djangoproject.com/ticket/23547), but it
+    # seems that since the tests and documentation are already written
+    # this behavior is Truth.
+    # see the note in the documentation
+    # https://docs.djangoproject.com/en/dev/ref/forms/fields/#booleanfield
+
+    _char_required = forms.CharField(required=True)
+    _char = forms.CharField(required=False)
+    _decimal = forms.DecimalField(max_digits=8, decimal_places=3)
+    _float = forms.FloatField()
+    _integer = forms.IntegerField()
+    _null_boolean = forms.NullBooleanField()
+
+
+class TemporalFieldForm(ReprForm):
+    _date = forms.DateField()
+    _date_time = forms.DateTimeField()
+    _duration = forms.DurationField()
+    _time = forms.TimeField()
+    _split_date_time = forms.SplitDateTimeField()
+
+
+class EmailFieldForm(ReprForm):
+    _email = forms.EmailField()
+
+
+class SlugFieldForm(ReprForm):
+    _slug = forms.SlugField()
+
+
+class URLFieldForm(ReprForm):
+    _url = forms.URLField()
+
+
+class RegexFieldForm(ReprForm):
+    _regex = forms.RegexField(regex=r"[A-Z]{3}\.[a-z]{4}")
```

Comment:
Found it!  `from_regex` has been passed a `str` pattern, so it generates `str` strings.  Then Django compares them to unicode internally, Python emits a warning, and our tests fail because of the `-Werror` flag.

Adding a `u`-prefix to the pattern should be sufficient to fix this.

Reply:
ezpz